### PR TITLE
add past events page

### LINKED ISSUE #36: Past event's section
### Description
Use the same frontend of currently the event's page to create a past event's page. Add a date range selector in the UI (use anything placeholder, will be updated once designers have the design for it ready). Make a request to the the events api with the date range (refer #35 ) . Default request of the page should be January 1 of the year to the current date.
 
### Acceptance Criteria
* A past event's page that by default loads the events from the January 1 to current date.
* ~~A date range placeholder, that allows to change the date range and makes a new request to the events api with the date range~~.

### User Story
For each past event, I want to view its volunteers' information so that I can have a record for past events, as well as can reflect and improve volunteer experience.

### PR DESCRIPTION
**Pull request by shwetamurali**
## PR Title/Tagline

What does this PR change and why?

### How to Test

1. Please add or build upon a testing card to the [Notion page](http://notion.so) and link it here

### Important Changes

- Database change / migration to run
- Environment config change
- Breaking API change

### Related Github Issues

- #[Number of Issue]
- Or, remember to link to an issue on the sidebar

### Related PRs

- #PRNUMBER

### Checklist

- [ ]  Database schema docs have been updated or are not necessary
- [ ]  Code follows design and style guidelines
- [ ]  Code is commented with doc blocks
- [ ]  Tests have been written and executed or are not necessary
- [ ]  Latest code has been rebased from base branch (usually `develop`)
- [ ]  Commits follow guidelines (concise, squashed, etc)
- [ ]  Github issues have been linked in relevant commits
- [ ]  Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### CODE REVIEW

File: src/components/Header.jsx [hunk 1]
```diff
@@ -240,6 +240,12 @@ const Header = () => {
               >
                 <Styled.PageLink>Events</Styled.PageLink>
               </Link>
+              <Link
+                href="/past-events"
+                selected={currPageMatches("/events")}
```
Comment (abhigyank) [inline src/components/Header.jsx]: Correction: `selected={currPageMatches("/past-events")}`

File: src/screens/Events/User/EventTable.jsx [hunk 2]
```diff
@@ -36,21 +36,6 @@ const EventTable = ({ events, loading, onRegister, onUnregister, user }) => {
...
-                        <span>Unregister</span>
-                      </Styled.Button>
-                    </>
-                  ) : (
-                    <Styled.Button onClick={() => onRegister(event)}>
-                      <Icon color="grey3" name="add" />
-                      <span>Sign up</span>
-                    </Styled.Button>
-                  )}
-                </td>
```
Comment (abhigyank) [inline src/screens/Events/User/EventTable.jsx]: Needs to be kept. I guess this was the temporary change for viewing volunteer view?

File: src/screens/Events/index.jsx [hunk 3]
```diff
@@ -6,6 +6,6 @@ import UserEventManager from "./User";
 const IS_ADMIN = true;
 
 const EventManagerSelector = (props) =>
-  props.isAdmin || IS_ADMIN ? <AdminEventManager /> : <UserEventManager />;
```
Comment (abhigyank) [inline src/screens/Events/index.jsx:9]: Same as above: Needs to be kept. I guess this was the temporary change for viewing volunteer view?

File: src/screens/PastEvents/index.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import AdminEventManager from "./Admin";
+import UserEventManager from "./User";
+
+// todo: delete once authorization has been added
+const IS_ADMIN = true;
+
+const EventManagerSelector = (props) =>
+  <UserEventManager />;
```
Comment (abhigyank) [inline src/screens/PastEvents/index.jsx]: `props.isAdmin || IS_ADMIN ? <AdminEventManager /> : <UserEventManager />;`

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+};
+
+const EventManager = () => {
+  const [loading, setLoading] = useState(true);
+  const [events, setEvents] = useState([]);
+  const [showCreateModal, setShowCreateModal] = useState(false);
+
+  const onRefresh = () => {
+    setLoading(true);
+    fetchEvents()
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: We need to fetch past events before today, so need to pass an end date to the function. `fetchEvents(undefined, new Date().toLocaleDateString("en-US"))`

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from "react";
+import styled from "styled-components";
+import { Button } from "reactstrap";
+import Icon from "../../../components/Icon";
+import EventTable from "./EventTable";
+import { fetchEvents } from "../../../actions/queries";
+import EventCreateModal from "./EventCreateModal";
+import EventEditModal from "./EventEditModal";
+import EventDeleteModal from "./EventDeleteModal";
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove these as we wont require to create, edit or delete event from the past-events page.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    background: white;
+    border: none;
+    color: black;
+  `,
+};
+
+const EventManager = () => {
+  const [loading, setLoading] = useState(true);
+  const [events, setEvents] = useState([]);
+  const [showCreateModal, setShowCreateModal] = useState(false);
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  };
+
+  const onCreateClicked = () => {
+    setShowCreateModal(true);
+  };
+
+  const toggleCreateModal = () => {
+    setShowCreateModal((prev) => !prev);
+    onRefresh();
+  };
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+  useEffect(() => {
+    onRefresh();
+  }, []);
+
+  const [showEditModal, setShowEditModal] = useState(false);
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+  const [currEvent, setCurrEvent] = useState(null);
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+    onRefresh();
+  };
+  const onDeleteClicked = (event) => {
+    setShowDeleteModal(true);
+    setCurrEvent(event);
+  };
+  const toggleDeleteModal = () => {
+    setShowDeleteModal((prev) => !prev);
+    onRefresh();
+  };
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+
+  return (
+    <Styled.Container>
+      <Styled.HeaderContainer>
+        <Styled.Button onClick={onCreateClicked}>
+          <Icon color="grey3" name="add" />
+          <span>Create</span>
+        </Styled.Button>
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+        <Styled.Button onClick={onRefresh}>
+          <Icon color="grey3" name="refresh" />
+          <span> Refresh</span>
+        </Styled.Button>
+      </Styled.HeaderContainer>
+      <EventTable
+        events={events}
+        loading={loading}
+        onEditClicked={onEditClicked}
+        onDeleteClicked={onDeleteClicked}
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventManager.jsx [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+      <EventEditModal
+        open={showEditModal}
+        toggle={toggleEditModal}
+        event={currEvent}
+      />
+      <EventDeleteModal
+        open={showDeleteModal}
+        toggle={toggleDeleteModal}
+        event={currEvent}
+      />
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventManager.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventTable.jsx [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+          </tr>
+        </thead>
+        <tbody>
+          {!loading &&
+            events.map((event, idx) => (
+              <EventTableRow
+                key={idx}
+                event={event}
+                onEditClicked={onEditClicked}
+                onDeleteClicked={onDeleteClicked}
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventTable.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventTable.jsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+      </Table.Table>
+      {loading && <Loading />}
+    </Table.Container>
+  );
+};
+EventTable.propTypes = {
+  loading: PropTypes.bool,
+  events: PropTypes.array,
+  onEditClicked: PropTypes.func,
+  onDeleteClicked: PropTypes.func,
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventTable.jsx]: Can remove.

File: src/screens/PastEvents/Admin/EventTableRow.jsx [hunk 17]
```diff
@@ -0,0 +1,86 @@
...
+      <td>
+        <Styled.Button onClick={() => onEditClicked(event)}>
+          <Icon color="grey3" name="create" />
+        </Styled.Button>
+      </td>
+      <td>
+        <Styled.Button onClick={() => onDeleteClicked(event)}>
+          <Icon color="grey3" name="delete" />
+        </Styled.Button>
+      </td>
```
Comment (abhigyank) [inline src/screens/PastEvents/Admin/EventTableRow.jsx]: Can remove.

**Review by abhigyank — CHANGES_REQUESTED** [review]

File: src/screens/Events/User/EventTable.jsx (hunk 2 above)
Comment (shwetamurali) [reply src/screens/Events/User/EventTable.jsx]: adding this back gives an error - the user ID is undefined

Comment (abhigyank) [reply src/screens/Events/User/EventTable.jsx]: Probably need to update the .env file that Jack has updated to support user logins. (It works for me with that, so should be fine to add back I guess).

Comment (abhigyank) [reply src/screens/Events/User/EventTable.jsx]: Or could be because, this branch doesn't have the changes Jack pushed in for fixing the auth.

**Comment by abhigyank** [comment]
Merged, good work @shwetamurali! Thanks :rocket:

**Pull request merged**